navapbc/wic-mt-demo-project-eligibility-screener
# Refactor <Trans> to support embedded urls that a content editor can update

## Ticket

https://wicmtdp.atlassian.net/browse/WMDP-252

## Changes
> What was added, updated, or removed in this PR.

- Add a new <TransLine> component that handles link interpolation using data in the translation file
- Convert all uses of <Trans> to <TransLine>
- Rename `i18nKey` to `I18nKey` for consistency

## Context for reviewers
> Testing instructions, background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers. Explain how the changes were verified.

This PR adds support to keep interpolated urls in the translation file next to the translation content by adding a `<TransLine>` component.

Any translation element that ends in `.text` is assumed to have a sibling `.link` that is an array of urls. For example:

```json
    "footer1": {
      "links": [
        "https://dphhs.mt.gov/ecfsd/wic/index",
        "https://www.signupwic.com/"
      ],
      "text": "This site is a demonstration project built with <0>Montana WIC</0> and is for example purposes only. If you need to submit an application for WIC, please use <1>SignUpWIC.com</1> to find a location near you."
    },
```

## Testing
> Screenshots, GIF demos, code examples or output to help show the changes working as expected. ProTip: you can drag and drop or paste images into this textbox.

First test: Run `yarn test` to run the new TransLine unit tests

Second test: Check that the links are still working as expected
  1. Checkout this branch
  2. Build and run the container: `docker-compose up -d --build`
  5. Navigate to localhost:6006
  6. Check the links in the footer to make sure they still link correctly (see screenshot below for an example)
  7. Remove the container when you're done: `docker-compose down` 

![Screen Shot 2022-12-05 at 4 04 17 PM](https://user-images.githubusercontent.com/67701/205742489-e91646cd-87b9-4a0e-a4df-c0a318fca81a.png)
